plugin-gradle/src/main/java/com/diffplug/gradle/spotless/TypescriptExtension.java

```diff
@@ -58,15 +58,15 @@ public TypescriptFormatExtension config(Map<String, Object> config) {
 
 		public TypescriptFormatExtension configFile(String filetype, String path) {
 			this.configFileType = requireNonNull(filetype);
-			this.configFilePath = requireNonNull(path);
+			this.configFilePath = path; // might be null for 'editorconfig'
```

Comment:
Does it find `.editorconfig` automatically?  This would break spotless' up-to-date checking.  Can we ask the user to specify the path to the `.editorconfig` manually?  If it's a big project, it's okay to merge as-is, but we should open a bug and reference it in the README to let users know that if they use `.editorconfig`, they might get up-to-date bugs from spotless.

Also:

- should `String filetype` be an enum?
- should `String path` be an `Object` that we resolve later with `project.file()`?

Reply:
The tsfmt options are inconsistent for the `editorconfig` param compared to the other config file types:

```typescript
export interface Options {
    dryRun?: boolean;
    verbose?: boolean;
    baseDir?: string;
    replace: boolean;
    verify: boolean;
    tsconfig: boolean;
    tsconfigFile: string | null;
    tslint: boolean;
    tslintFile: string | null;
    editorconfig: boolean;
    vscode: boolean;
    vscodeFile: string | null;
    tsfmt: boolean;
    tsfmtFile: string | null;
}
```
([Source](https://github.com/vvakame/typescript-formatter/blob/7764258ad42ac65071399840d1b8701868510ca7/lib/index.ts#L30))

So we could let the user specify a file location but we cannot pass it to tsfmt... one option is to just remove this option from our interface, there are plenty of other options available.

Speaking of the other options: I think it might be necessary to really read the content of the config files passed to tsfmt (using `FileSignature`) in order to support up-to-date checks. Currently I only pass the paths to tsfmt, but I don't look at the content of them. So if the content changes, but not the path, gradle would wrongly assume that nothing has changed. Does spotless already have similar cases (to use consistent solutions to similar problems...).

> should `String filetype` be an enum?

We could make it one. Or we provide specific methods in the extension like `tslintConfig('/path/to/tslint.json')`, `vscodeConfig('/path/to/vscode.json')` and so on... 

> should `String path` be an `Object` that we resolve later with `project.file()`?

We definitely could do that, yes!

Reply:
working on making the config file an `Object`:

```java
public void configFile(String filetype, Object configFilePath) {
...
}
```

but keep getting

> No signature of method: com.diffplug.gradle.spotless.TypescriptExtension$TypescriptFormatExtension.configFile() is applicable for argument types: (java.lang.String, java.io.File) 

when running integration test in the form
```groovy
...
tsfmt().configFile('tsfmt', 'path/to/tsfmt.json')
...
```

Any idea what I'm doing wrong?